docs/adr/0001-ubuntu-24-collection-selection-for-puppet-agent-install-task.md

```diff
@@ -0,0 +1,54 @@
+# 1. Ubuntu 24 Collection Selection for puppet_agent::install Task
+
+Date: 2025-04-23
+
+## Status
+
+Proposed
+
+## Context
+
+When running Bolt apply with Ubuntu 24 (codename: noble), installations were failing because the puppet_agent::install task defaulted to using the generic 'puppet' collection, resulting in the installation of `puppet-release-noble.deb` instead of the version-specific `puppet8-release-noble.deb`.
+
+This issue specifically affected Ubuntu 24, while other operating systems worked correctly. The underlying cause was that the generic 'puppet' repository didn't properly support Ubuntu 24, whereas the version-specific repositories (like 'puppet8') did have proper packages available.
+
+The puppet_agent::install task (implemented in install_shell.sh) provides a mechanism for installing the Puppet agent across various platforms. It accepts parameters including 'collection', which determines which package collection to use. If no collection is specified, it was defaulting to the generic 'puppet' collection.
+
+A temporary fix was implemented in Bolt's apply_prep.rb function, which added logic to determine the Puppet version and set the collection parameter accordingly. While this fix worked, it violated architectural separation of concerns by placing target-specific logic in the higher-level orchestration function. For more information, see [this pr](https://github.com/puppetlabs/bolt/pull/3376) and [this PR](https://github.com/puppetlabs/bolt-private/pull/16).
+
+## Decision
+
+Move the Puppet version detection and collection selection logic from Bolt's apply_prep function to the puppet_agent::install task's install_shell.sh script, which is the architecturally appropriate location for this functionality.
+
+The implementation enhances the collection selection logic to:
+
+1. First check if a collection was explicitly specified
+   - If so, use that collection as before
+   - This maintains backward compatibility and user control
+
+2. If no collection is specified, intelligently determine the appropriate collection:
```

Comment:
What's the usecase for people calling the install task on nodes where /opt/puppetlabs/puppet/VERSION exists already?

Reply:
Hi @bastelfreak.  This is a draft to do some testing against.  I've added your question to the outstanding tasks in the PR description.